src/connector.ts

```diff
@@ -52,29 +55,40 @@ export declare interface DriverOptions {
 // adds extra logic to async initialize items.
 class CloudSQLInstanceMap extends Map {
   async loadInstance({
-    type,
+    ipType,
+    authType,
     instanceConnectionName,
     sqlAdminFetcher,
   }: {
-    type: IpAdressesTypes;
+    ipType: IpAdressesTypes;
+    authType: AuthTypes;
     instanceConnectionName: string;
     sqlAdminFetcher: SQLAdminFetcher;
   }): Promise<void> {
     // in case an instance to that connection name has already
     // been setup there's no need to set it up again
-    if (this.has(instanceConnectionName)) {
+    if (this.has(this.getKey({instanceConnectionName, ipType, authType}))) {
```

Comment:
This should be a separate PR as it's a departure from other connectors.

Reply:
curious, how does the other connectors handle possible concurrent ip/auth types within a single instance?

Reply:
They don't handle it. The direction here is correct, but I'd like to land this change in Go first and with some thought into the implications of this change.

Reply:
python connector handles this by throwing an error if `enable_iam_auth` argument has changed since last `connect` call. https://github.com/GoogleCloudPlatform/cloud-sql-python-connector/blob/main/google/cloud/sql/connector/connector.py#L188-L197

Reply:
This is a good example of where we need to bring all the connectors in line feature-wise. But again not in this PR. ;-)

Reply:
I have the impression the follow up PRs are going to be quite small once we remove this and split the proposed changes appart 😊

Reply:
Little follow up -- I think we want to be following Python here. But still probably better as a separate PR after a one-pager looking at all the connectors.